spiped/main.c

```diff
@@ -134,16 +135,15 @@ main(int argc, char * argv[])
 			opt_k = optarg;
 			break;
 		GETOPT_OPTARG("-n"):
-			if (opt_n != 0)
+			if (opt_n_set)
 				usage();
+			opt_n_set = 1;
 			if ((opt_n = strtoimax(optarg, NULL, 0)) == 0) {
```

Comment:
This does not permit the user to set `-n 0`.

Unfortunately, we can't rely on errno; "If no conversion could be performed, zero shall be returned [CX] ￼  and errno may be set to [EINVAL]. "
http://pubs.opengroup.org/onlinepubs/9699919799/functions/strtoimax.html

The man page on my Ubuntu machine says "If nothing was  found  to convert, zero is returned.  On overflow or underflow INTMAX_MAX or INTMAX_MIN or UINTMAX_MAX is returned, and errno is set to ERANGE." so POSIX wasn't kidding when they wrote that *errno* **may** be set.